tests/console/nvme_checks.pm

```diff
@@ -0,0 +1,121 @@
+    my $MODE       = '\/dev\/' . $nvm_test_data->{nvm_char_device};
+    my $SN         = $nvm_test_data->{sn};
+    my $MODEL      = $nvm_test_data->{model};
+    my $NAMESPACE  = $nvm_test_data->{namespace_count};
+    my $DISK_SIZE  = $nvm_test_data->{nvm_disk_size};
+    my $BLOCK_SIZE = $nvm_test_data->{nvm_block_size};
+    my $list_regex = qr/${MODE}\s+$SN\s+${MODEL}\s+${NAMESPACE}\s+${DISK_SIZE}\s+GB.*\d+\s+GB\s+${BLOCK_SIZE}/;
+    validate_script_output("nvme list",                                                                     sub { $list_regex });
+    validate_script_output("nvme list-ns /dev/$nvm_test_data->{nvm_char_device} | awk -F ':' '{print $2}'", sub { /$nvm_test_data->{nvm_ns}/ });
+    # the details of this namespace
+    # check LBA Format fields
+    # ms = Metadata Size, 0 means metadata is not supported
+    # lbads = LBA Data Size, in terms of a power of two, so LBA data size is 2⁹ = 512 bytes
+    # rp = Relative Performance, the lower the better
+    validate_script_output("nvme id-ns /dev/$nvm_test_data->{nvm_disk} --namespace-id=$nvm_test_data->{nvm_ns}", sub { m/lbaf\s+0\s+:\s+ms:0\s+lbads:9\s+rp:0\s\(in use\)/ });
+    # show registers
+    my $ret_code = script_run("nvme show-regs -H /dev/$nvm_test_data->{nvm_disk}");
+    record_soft_failure "bsc#1175217 - not permitted shared memory access to get registers" if ($ret_code > 0);
+}
+
+sub _issue_read_command_and_compare_with_dd {
+    my $nvm_test_data = shift;
+    assert_script_run("dd if=/dev/$nvm_test_data->{nvm_disk} of=lba.0.dd bs=512 count=1");
+    # read LBA 0 through nvme utility
+    assert_script_run("nvme read /dev/$nvm_test_data->{nvm_disk} --start-block=0 --block-count=0 --data-size=512 --data=lba.0.read");
+    # Read is opcode 0x02, we are sending command to namespace 0x1, reading 512 bytes
+    # code word 10 and 11 (cdw10 and cdw11) specifies the start of reading in LBA blocks,
+    # and the Bits 15:00 of code word 12 (cdw12) is number of blocks to read (set to 0 to indicate 1 block will be read).
+    assert_script_run("nvme io-passthru /dev/$nvm_test_data->{nvm_char_device} --opcode=0x02 --namespace-id=0x1 --data-len=512 --read --cdw10=0 --cdw11=0 --cdw12=0 -b > lba.0.io");
+    assert_script_run("cmp lba.0.dd lba.0.read", fail_message => "The Admin read command is broken. Results are different between the helper read command and the manual dd");
+    assert_script_run("cmp lba.0.dd lba.0.io", fail_message => "The Submission and Completion queue for Admin commands differs from I/O commands");
+}
+
+sub _check_if_nvme_tools_is_installed {
+    my $ret = zypper_call('se -x -i nvme-cli', exitcode => [0, 104]);
+    if ($ret == 104) {
+        record_soft_failure "bsc#1172866 - nvme-cli is not installed" unless script_run("rpm -qa nvme-cli");
+        zypper_call("in nvme-cli");
+    }
+    record_info "nvme_cli found", "No installation is needed for nvme_cli" if $ret == 0;
+}
```

Comment:
Here, return code 0 means that package is found in database, not that it is installed.

Reply:
So the only solution from my POV is to check the output of zypper. I refactor the code  accordingly to check for the `+i`.

Reply:
if/else would work better here than run the regex twice (but better replace the whole block by suggested one)